Duplicati/UnitTest/PurgeTesting.cs

```diff
@@ -146,7 +146,7 @@ public void PurgeTest()
                 Console.WriteLine("Files:");
                 Console.WriteLine("  " + string.Join(Environment.NewLine + "  ", files.Select(x => string.Format("{0}: {1}", x.Path, string.Join(" - ", x.Sizes.Select(y => y.ToString()))))));
 
-                Assert.AreEqual(4, filesets.Length, "Incorrect number of filesets after final backup");
+                Assert.AreEqual(1, filesets.Length, "Incorrect number of filesets after final backup");

```

Comment:
Sorry, I'm not familiar with the expectations here.  Does this mean that a purge operation marks the existing backup sets as partial?

Reply:
The partial backups are removed when the next most recent full backup is removed. The 4 was from when all the backups were not differentiated.

Reply:
I'm not super familiar with this test, but why are partials appearing?  My understanding of a purge operation is that a user has decided that a file shouldn't have been included in the backup sets (e.g., for security reasons) and hence that file should be removed from all backup sets.  However, I think these are still considered by the user to be "full" backups.

Reply:
The test is checking for the fileset count and is not looking at particular files in the fileset.

Reply:
Forgive my confusion, but I'm still trying to understand this.  When running this test, I notice that
1. 3 backups are run.  After this, `c.List` shows 3 filesets and indicates that they are full backups.
2. A purge is run on a file that exists in all filesets.  After this, `c.List` shows 3 filesets and indicates that they are partial backups.

In step 2, I would have expected the result of `c.List` to have 3 filesets (each with one less file than previously), but still marked as full backups.  Is my intuition incorrect?  Why are these now considered partial after a purge?

Reply:
I think you're right. They should be full backups. It looks like purge creates new filesets. I need to have it carry over the isFullBackup value when recreating a fileset.

Reply:
PurgeFilesHandler re-creates filesets the value of isFullBackup was then always set to partial. The value of isFullBackup is now carried over by PurgeFilesHandler.

I checked PurgeBroken and it is fine since it calls PurgeFilesHandler.

Commits are in.